pkg/utils/files.go

```diff
@@ -47,9 +50,79 @@ func BuildFileList(paths []string) ([]string, error) {
 	return fileList, nil
 }
 
+func BuildFolderList(paths []string) ([]string, error) {
```

Comment:
We're mixing terminology - can you refactor to "directory" rather than "folder" in this function?